libraries/elkstack.rb

```diff
@@ -0,0 +1,40 @@
+# Module for reusable code for elkstack
+module Elkstack
+  def self.get_platformstack_elkstack_elklogging(current_node, *keys)
+    require 'chef/sugar'
+
+    deprecated_key = ['platformstack', 'elkstack_logging', keys].flatten
+    deprecated_value = current_node.deep_fetch(*deprecated_key)
+
+    current_key = ['elkstack', 'logging', keys].flatten
+    current_value = current_node.deep_fetch(*current_key)
+
+    if deprecated_value && !current_value
+      Chef::Log.warn("You are using deprecated attribute keys under node['platformstack']['elkstack_logging']: #{keys.join(',')}")
+    elsif deprecated_value && current_value
+      Chef::Log.warn("You are using deprecated attribute keys and have non-deprecated keys set at the same time (will use non-deprecated keys first): #{keys.join(',')}")
+    end
+
+    # return the preferred value first, followed by the deprecated one
+    current_value || deprecated_value
+  end
+
+  def self.get_platformstack_elkstack_monitoring(current_node, *keys)
```

Comment:
This looks remarkably similar to the method just above it. Maybe we can write a generic one?